[api client connector] integration with basic authentication crashes

## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
Simple integration (select from db, create task from todo connector):

![screenshot_20190624_150708](https://user-images.githubusercontent.com/14313995/60021165-bf92bd00-9691-11e9-915a-2824abbddff3.png)

I am using our todo app, to create custom api connector from url you can use:
https://raw.githubusercontent.com/syndesisio/syndesis-qe/master/ui-tests/src/test/resources/swagger/connectors/todo.swagger.yaml

The integration goes into crashloopbackoff with following error in its log:


```
ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application startup failed
org.apache.camel.spring.boot.CamelSpringBootInitializationException: java.lang.IllegalArgumentException: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.basic:username_password
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:250) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:57) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_191]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_191]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_191]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_191]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
Caused by: java.lang.IllegalArgumentException: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.basic:username_password
	at io.syndesis.integration.runtime.sb.IntegrationRuntimeAutoConfiguration$1.beforeApplicationStart(IntegrationRuntimeAutoConfiguration.java:95) ~[integration-runtime-springboot-1.7.3.jar!/:1.7.3]
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:152) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	... 24 common frames omitted
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.basic:username_password
	at io.syndesis.integration.runtime.handlers.ConnectorStepHandler.handle(ConnectorStepHandler.java:146) ~[integration-runtime-1.7.3.jar!/:1.7.3]
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.configureFlow(IntegrationRouteBuilder.java:222) ~[integration-runtime-1.7.3.jar!/:1.7.3]
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.configure(IntegrationRouteBuilder.java:139) ~[integration-runtime-1.7.3.jar!/:1.7.3]
	at org.apache.camel.builder.RouteBuilder.checkInitialized(RouteBuilder.java:462) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.builder.RouteBuilder.configureRoutes(RouteBuilder.java:402) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(RouteBuilder.java:383) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1027) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1024) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext.addRoutes(DefaultCamelContext.java:1024) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at io.syndesis.integration.runtime.sb.IntegrationRuntimeAutoConfiguration$1.beforeApplicationStart(IntegrationRuntimeAutoConfiguration.java:93) ~[integration-runtime-springboot-1.7.3.jar!/:1.7.3]
	... 25 common frames omitted
Caused by: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.basic:username_password
	at java.lang.Enum.valueOf(Enum.java:238) ~[na:1.8.0_191]
	at io.syndesis.connector.rest.swagger.AuthenticationType.valueOf(AuthenticationType.java:20) ~[connector-rest-swagger-1.7.3.jar!/:1.7.3]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.lambda$customize$0(AuthenticationCustomizer.java:33) ~[connector-rest-swagger-1.7.3.jar!/:1.7.3]
	at io.syndesis.integration.component.proxy.ComponentCustomizer.consumeOption(ComponentCustomizer.java:43) ~[integration-component-proxy-1.7.3.jar!/:1.7.3]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.customize(AuthenticationCustomizer.java:32) ~[connector-rest-swagger-1.7.3.jar!/:1.7.3]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.customize(AuthenticationCustomizer.java:28) ~[connector-rest-swagger-1.7.3.jar!/:1.7.3]
	at io.syndesis.integration.runtime.handlers.HandlerCustomizer.customizeComponent(HandlerCustomizer.java:67) ~[integration-runtime-1.7.3.jar!/:1.7.3]
	at io.syndesis.integration.runtime.handlers.ConnectorStepHandler.handle(ConnectorStepHandler.java:128) ~[integration-runtime-1.7.3.jar!/:1.7.3]
	... 35 common frames omitted
```

API key causes this as well
```
2019-06-25 06:49:53.103 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application startup failed
org.apache.camel.spring.boot.CamelSpringBootInitializationException: java.lang.IllegalArgumentException: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.apiKey:api_key
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:250) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:57) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_191]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_191]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_191]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_191]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
Caused by: java.lang.IllegalArgumentException: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.apiKey:api_key
	at io.syndesis.integration.runtime.sb.IntegrationRuntimeAutoConfiguration$1.beforeApplicationStart(IntegrationRuntimeAutoConfiguration.java:95) ~[integration-runtime-springboot-1.7.4.jar!/:1.7.4]
	at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent(RoutesCollector.java:152) ~[camel-spring-boot-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	... 24 common frames omitted
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.apiKey:api_key
	at io.syndesis.integration.runtime.handlers.ConnectorStepHandler.handle(ConnectorStepHandler.java:146) ~[integration-runtime-1.7.4.jar!/:1.7.4]
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.configureFlow(IntegrationRouteBuilder.java:222) ~[integration-runtime-1.7.4.jar!/:1.7.4]
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.configure(IntegrationRouteBuilder.java:139) ~[integration-runtime-1.7.4.jar!/:1.7.4]
	at org.apache.camel.builder.RouteBuilder.checkInitialized(RouteBuilder.java:462) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.builder.RouteBuilder.configureRoutes(RouteBuilder.java:402) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(RouteBuilder.java:383) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1027) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1024) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at org.apache.camel.impl.DefaultCamelContext.addRoutes(DefaultCamelContext.java:1024) ~[camel-core-2.21.0.fuse-740028.jar!/:2.21.0.fuse-740028]
	at io.syndesis.integration.runtime.sb.IntegrationRuntimeAutoConfiguration$1.beforeApplicationStart(IntegrationRuntimeAutoConfiguration.java:93) ~[integration-runtime-springboot-1.7.4.jar!/:1.7.4]
	... 25 common frames omitted
Caused by: java.lang.IllegalArgumentException: No enum constant io.syndesis.connector.rest.swagger.AuthenticationType.apiKey:api_key
	at java.lang.Enum.valueOf(Enum.java:238) ~[na:1.8.0_191]
	at io.syndesis.connector.rest.swagger.AuthenticationType.valueOf(AuthenticationType.java:20) ~[connector-rest-swagger-1.7.4.jar!/:1.7.4]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.lambda$customize$0(AuthenticationCustomizer.java:33) ~[connector-rest-swagger-1.7.4.jar!/:1.7.4]
	at io.syndesis.integration.component.proxy.ComponentCustomizer.consumeOption(ComponentCustomizer.java:43) ~[integration-component-proxy-1.7.4.jar!/:1.7.4]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.customize(AuthenticationCustomizer.java:32) ~[connector-rest-swagger-1.7.4.jar!/:1.7.4]
	at io.syndesis.connector.rest.swagger.AuthenticationCustomizer.customize(AuthenticationCustomizer.java:28) ~[connector-rest-swagger-1.7.4.jar!/:1.7.4]
	at io.syndesis.integration.runtime.handlers.HandlerCustomizer.customizeComponent(HandlerCustomizer.java:67) ~[integration-runtime-1.7.4.jar!/:1.7.4]
	at io.syndesis.integration.runtime.handlers.ConnectorStepHandler.handle(ConnectorStepHandler.java:128) ~[integration-runtime-1.7.4.jar!/:1.7.4]
	... 35 common frames omitted
```

--- a/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/AuthenticationCustomizer.java
+++ b/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/AuthenticationCustomizer.java
@@ -30,7 +30,8 @@ public class AuthenticationCustomizer implements ComponentProxyCustomizer {
     @Override
     public void customize(final ComponentProxyComponent proxyComponent, final Map<String, Object> options) {
         consumeOption(options, "authenticationType", authenticationTypeObject -> {
-            final AuthenticationType authenticationType = AuthenticationType.valueOf(String.valueOf(authenticationTypeObject));
+            final String authenticationTypeAsString = String.valueOf(authenticationTypeObject);
+            final AuthenticationType authenticationType = AuthenticationType.fromString(authenticationTypeAsString);
             if (authenticationType == AuthenticationType.none) {
                 return;
             }

--- a/app/connector/rest-swagger/src/test/java/io/syndesis/connector/rest/swagger/AuthenticationCustomizerTest.java
+++ b/app/connector/rest-swagger/src/test/java/io/syndesis/connector/rest/swagger/AuthenticationCustomizerTest.java
@@ -17,6 +17,7 @@ package io.syndesis.connector.rest.swagger;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Stream;
 
 import io.syndesis.integration.component.proxy.ComponentProxyComponent;
 
@@ -147,6 +148,40 @@ public class AuthenticationCustomizerTest {
         assertThat(options).doesNotContainKeys("authenticationType", "accessToken", "refreshToken", "authorizationEndpoint");
 
         assertAuthorizationHeaderSetTo(component, "Bearer new-access-token");
+    }
+
+    @Test
+    public void shouldSupportAllAuthenticatioValues() {
+        final ComponentProxyComponent component = new SwaggerProxyComponent("test", "test");
+        final CamelContext context = mock(CamelContext.class);
+        component.setCamelContext(context);
+
+        final AuthenticationCustomizer customizer = new AuthenticationCustomizer();
+
+        Stream.of(AuthenticationType.values()).forEach(authenticationType -> {
+            final Map<String, Object> options = new HashMap<>();
+            options.put("authenticationType", authenticationType.name());
+
+            customizer.customize(component, options);
+            // no IllegalStateException thrown
+        });
+    }
+
+    @Test
+    public void shouldSupportNamedAuthenticatioValues() {
+        final ComponentProxyComponent component = new SwaggerProxyComponent("test", "test");
+        final CamelContext context = mock(CamelContext.class);
+        component.setCamelContext(context);
+
+        final AuthenticationCustomizer customizer = new AuthenticationCustomizer();
+
+        Stream.of(AuthenticationType.values()).forEach(authenticationType -> {
+            final Map<String, Object> options = new HashMap<>();
+            options.put("authenticationType", authenticationType.name() + ":name");
+
+            customizer.customize(component, options);
+            // no IllegalStateException thrown
+        });
     }
 
     private static void assertAuthorizationHeaderSetTo(final ComponentProxyComponent component, final String value) throws Exception {

